cdap-unit-test/src/main/java/co/cask/cdap/test/internal/DefaultServiceManager.java

```diff
@@ -103,8 +112,23 @@ public URL getServiceURL() {
   @Override
   public URL getServiceURL(long timeout, TimeUnit timeoutUnit) {
     String discoveryName = String.format("service.%s.%s.%s", accountId, applicationId, serviceName);
-    String path = String.format("%s/namespaces/%s/apps/%s/services/%s/methods/",
+    ServiceDiscovered discovered = discoveryServiceClient.discover(discoveryName);
+    return createURL(new RandomEndpointStrategy(discovered).pick(timeout, timeoutUnit), applicationId, serviceName);
+  }
+
+  private URL createURL(@Nullable Discoverable discoverable, String applicationId, String serviceName) {
```

Comment:
Also, is there a utility class where we could share this ``createURL`` method between main code and test code, although I guess the point of doing this refactor was also to remove the static utility class ``ServiceDiscoveries``? Not sure if it would be a good idea to keep that class with just this method?